language/move-vm/runtime/src/loader.rs

```diff
@@ -126,12 +127,15 @@ impl ScriptCache {
 /// replaced with indices into these global structures in loaded modules.  All access to the
 /// ModuleCache via the Loader is under an RWLock.
 pub struct ModuleCache {
-    /// Compiled modules go in this cache once they have been verified (individually and linkage
-    /// with dependencies and friends)
+    /// Compiled modules go in this cache once they have been individually verified.
     compiled_modules: BinaryCache<ModuleId, CompiledModule>,
-    /// Loaded modules go in this cache once their structs and functions have populated `structs`
-    /// and `functions` below.
-    loaded_modules: BinaryCache<ModuleId, Module>,
+    /// Loaded modules go in this cache once their compiled modules have been link-checked, and
+    /// structs and functions have populated `structs` and `functions` below.
+    ///
+    /// The `AccountAddress` in the key is the "link context", and the `ModuleId` is the ID of the
+    /// module whose load was requested. A mapping `(ctx, id) => module` means that when `id` was
+    /// requested in context `ctx`, `module` was loaded.
+    loaded_modules: BinaryCache<(AccountAddress, ModuleId), LoadedModule>,
```

Comment:
That's not a bad idea -- i.e. make the `DataStore` responsible for this cache, and then you don't have to worry about threading the link context through to be used as a cache key here (really that's the only reason to do it).

The only thing that gives me pause is that while that works well for the current system, I'm not sure it works well when you want to go down the dependency Merkle hash approach (because that will also affect this key).  I'll think on it.